lib/string_reverse.rb

```diff
@@ -1,4 +1,22 @@
+
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  my_string << "not implemented"
+  # my_string << "not implemented"
+  if my_string == nil
+    return nil
+  end
+
+  i = 1
+  new_string = []

```

Comment:
Whilst this approach works, because you're using a new string, your space complexity will be O(n). If you follow the algorithm we discussed in class, the space complexity will be O(1).  Take a read through the slides we used in class: https://drive.google.com/file/d/0B__DV26QHsH4eHJqTWttLUdNZk0/view?usp=sharing and slack or email me if you have any questions.